web_poet/serialization/functions.py

```diff
@@ -0,0 +1,152 @@
+import json
+from typing import Any, Dict, Generic, Type, Union
+
+import attrs
+
+from .. import (
+    HttpResponse,
+    HttpResponseBody,
+    HttpResponseHeaders,
+    Injectable,
+    ResponseUrl,
+)
+from ..page_inputs.url import _Url
+from .api import SerializedData, deserialize, register_serialization, serialize
+
+
+def _serialize_dict(o: dict) -> SerializedData:
+    return {"json": json.dumps(o).encode()}
+
+
+def _deserialize_dict(t: Type[dict], data: SerializedData) -> dict:
+    return t(json.loads(data["json"].decode()))
```

Comment:
```suggestion
    return t(json.loads(data["json"]))
```

json.loads supports bytes directly